roles/ec2_networking_resources/tasks/create.yml

```diff
@@ -0,0 +1,80 @@
+---
+- name: Validate options
+  ansible.builtin.fail:
+    msg: "When creating resources, all of the following options must be provided: ec2_networking_resources_vpc_cidr_block, ec2_networking_resources_subnet_cidr_block, ec2_networking_resources_sg_internal_name"
+  when: ec2_networking_resources_vpc_cidr_block | default("", true) == "" or
+    ec2_networking_resources_subnet_cidr_block | default("", true) == "" or
+    ec2_networking_resources_sg_internal_name | default("", true) == ""
+
+- name: Create VPC
+  amazon.aws.ec2_vpc_net:
+    name: "{{ ec2_networking_resources_vpc_name }}"
+    cidr_block: "{{ ec2_networking_resources_vpc_cidr_block }}"
+  register: ec2_networking_resources_vpc_result
+
+- name: Set VPC ID
+  ansible.builtin.set_fact:
+    vpc_id: "{{ ec2_networking_resources_vpc_result.vpc.id }}"
+
+- name: Create VPC subnet
+  amazon.aws.ec2_vpc_subnet:
+    vpc_id: "{{ ec2_networking_resources_vpc_result.vpc.id }}"
```

Comment:
If you've already set the vpc id as a fact, I'd just use it throughout to be consistent. The same goes for the other cases here where you've set a variable. Either that, or remove the set_fact tasks, because it doesn't seem like those are used anywhere.

Reply:
Yeah, we need to set the `subnet_id` fact because it gets used by the `ec2_instance_create_delete` role when they're combined but I don't think the other facts are needed.